src/chart/generateCategoricalChart.tsx

```diff
@@ -662,16 +664,26 @@ const getAxisNameByLayout = (layout: LayoutType) => {
  * @return {Object} The offset of main part in the svg element
  */
 const calculateOffset = (
-  { props, graphicalItems, xAxisMap = {} as BaseAxisProps, yAxisMap = {} as BaseAxisProps }: any,
-  prevLegendBBox?: any,
-) => {
+  {
+    props,
+    graphicalItems,
+    xAxisMap = {},
+    yAxisMap = {},
+  }: {
+    props: CategoricalChartProps;
+    graphicalItems: Array<ReactElement>;
+    xAxisMap?: { [axisId: string]: XAxisProps };
+    yAxisMap?: { [axisId: string]: YAxisProps };
```

Comment:
nit: could we use `Record<string, X/YAxisProps>` instead?

Reply:
I chose the indexer because Record does not allow me to give a name to the key. Is there another difference between Record and object I am not aware of?

Reply:
No it's the same, I just personally prefer it for brevity but your reason makes sense